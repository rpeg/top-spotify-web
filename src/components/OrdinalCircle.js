import React from 'react';
import PropTypes from 'prop-types';

const OrdinalCircle = ({ position }) => (
  <div
    style={{
      position: 'absolute',
      zIndex: '25',
      top: '-10px',
      left: '-10px',
      width: '20px',
      height: '20px',
      fontSize: '12px',
      fontWeight: 'bold',
      border: '1px solid white',
      borderRadius: '12px',
      textAlign: 'center',
      background: '#191414',
    }}
  >
    {position}
  </div>
);

OrdinalCircle.propTypes = {
  position: PropTypes.number.isRequired,
};

export default OrdinalCircle;
