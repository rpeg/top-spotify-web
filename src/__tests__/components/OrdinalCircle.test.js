import React from 'react';
import { shallow } from 'enzyme';

import OrdinalCircle from '../../components/OrdinalCircle';

describe('OrdinalCircle', () => {
  const position = 1;

  it('should not log errors in console', () => {
    const wrapper = shallow(<OrdinalCircle position={position} />);
    const spy = jest.spyOn(global.console, 'error');
    expect(wrapper).not.toBeNull();
    expect(spy).not.toHaveBeenCalled();
  });

  it('renders correctly', () => {
    const wrapper = shallow(<OrdinalCircle position={position} />);
    expect(wrapper).toMatchSnapshot();
  });
});
