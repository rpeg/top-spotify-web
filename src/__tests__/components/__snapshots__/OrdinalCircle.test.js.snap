// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrdinalCircle renders correctly 1`] = `
<div
  style={
    Object {
      "background": "#191414",
      "border": "1px solid white",
      "borderRadius": "12px",
      "fontSize": "12px",
      "fontWeight": "bold",
      "height": "20px",
      "left": "-10px",
      "position": "absolute",
      "textAlign": "center",
      "top": "-10px",
      "width": "20px",
      "zIndex": "25",
    }
  }
>
  1
</div>
`;
